specification-derive/src/ast.rs

```diff
@@ -84,24 +88,37 @@ impl<'a> Variant<'a> {
                     original: &attr,
                     tokens: &attr.tokens,
                 }));
-            } 
+            } else if attr.path.is_ident("parent") {
+                if parent_attr.is_some() {
+                    return Err(Error::new_spanned(node, "duplicate #[parent()] attribute"));
+                }
+                let ident = attr.parse_args::<syn::Ident>().map_err(|err| Error::new(err.span(), "#[parent()] must be Spec variant name"))?;
+                if node.ident == ident {
+                    return Err(Error::new_spanned(node, "#[parent()] cannot be self"))
+                }
+                // take from set to keep proper span in case of errors
```

Comment:
Is it possible to also validate that the specified "parent" is a `Master` element?  It would be nice to validate that as part of this macro.

Reply:
good idea